### PR TITLE
feat: add option to return reason when is_alive returns false

### PR DESCRIPTION
**Pull request by olcai**
We want the caller to is_alive to be able to log the reason for why is_alive returns false. This is now possible by calling influx:is_alive(Client, true).

### CODE REVIEW

**Comment by olcai** [comment]
I'm not sure what the release procedures are for this repo. Please tell me what else I need to do.

File: src/influxdb_http.erl [hunk 1]
```diff
@@ -39,13 +39,14 @@ is_alive(v2, Client = #{headers := Headers}) ->
...
             {ok, 204, _, _} ->
                 true;
-            _ ->
-                false
+            Return ->
+                maybe_return_reason(Return, ReturnReason)
         end
-    catch _E:_R:_S ->
+    catch E:R:S ->
+        logger:error("[InfluxDB] is alive: ~0p ~0p ~0p", [E, R, S]),
```
Comment (zhongwencool) [inline src/influxdb_http.erl:46]:
```suggestion
        logger:error("[InfluxDB] is broken: ~0p ~0p ~0p", [E, R, S]),
```

Comment (olcai) [reply src/influxdb_http.erl]: I have now clarified that it is the function `is_alive` that's intended here, not that it is actually alive.

File: src/influxdb_http.erl [hunk 2]
```diff
@@ -55,11 +56,11 @@ is_alive(v1, Client) ->
...
                 true;
-            _ ->
-                false
+            Return ->
+                maybe_return_reason(Return, ReturnReason)
         end
     catch E:R:S ->
-        logger:error("[InfluxDB] is alive: ~0p ~0p ~0p", [E, R, S]),
+        logger:error("[InfluxDB] is_alive exception: ~0p ~0p ~0p", [E, R, S]),
         false
```
Comment (zmstone) [inline src/influxdb_http.erl:64]: maybe just return `{false, #{exception => E, reason => R, stacktrace => S}}` here.
caller should log this `false` reason.

Comment (zmstone) [reply src/influxdb_http.erl:64]: maybe add a `log_or_return(#{exception => E, reason => R, stacktrace => S}, ReturnReason)`.

Comment (olcai) [reply src/influxdb_http.erl]: Yes, that is smart. I have implemented it now.

**Review by thalesmg — COMMENTED** [review]
LGTM

For the release, I think we'll need to bump the version here in `app.src`, and will probably need a couple new `.appup.src` entries.

One tip is to create a temporary tag at this branch's HEAD (e.g.: `1.1.8-erik0`), use that version in EMQX and either run `scripts/update-appup.sh emqx-enterprise` locally or create a PR with that version and CI will help.

After the appup is ironed out, then we tag `master` with the true version.

**Review by zmstone — APPROVED** [review]

**Review by zmstone — CHANGES_REQUESTED** [review]
Sorry, please update .app.src and .appup.src as @thalesmg suggested.

File: src/influxdb.app.src [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {application,influxdb,
              [{description,"InfluxDB Client"},
-              {vsn,"git"},
+              {vsn,"1.1.8"},
```
[the remark this answers is not in the document]
Comment (olcai) [reply src/influxdb.app.src]: Okay, I've put it back to `git` again.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**